### PR TITLE
code challenge on kth from the end complete, along with multiple tests

### PR DESCRIPTION
**Pull request by SharinaS**
- [x] Top-level README “Table of Contents” is updated
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
     - [x] “Happy Path” - Expected outcome
     - [x] Expected failure
     - [x] Edge Case (if applicable/obvious)
 - [x] README for this challenge is complete
     - [x] Summary, Description, Approach & Efficiency, Solution
     - [x] Link to code
     - [x] Picture of whiteboard

### CODE REVIEW

File: code401Challenges/src/main/java/code401Challenges/linkedlist/LinkedList.java [hunk 1]
```diff
@@ -109,5 +109,28 @@ public Boolean includes(int value) {
...
+        Node curr = head;
+        int listLength = 0;
+
+        while (curr != null) {
+            listLength += 1;
+            curr = curr.next;
+        }
+        int stepsToTake = listLength - k;
+        if(stepsToTake < 0 || stepsToTake > listLength)
+            throw new IllegalArgumentException("k is too big, given the length of the list");
```
Comment (jeffborda) [inline code401Challenges/src/main/java/code401Challenges/linkedlist/LinkedList.java]: I love that you're throwing an IllegalArgumentException here! Double check the message it gives.

File: code401Challenges/src/test/java/code401Challenges/linkedlist/LinkedListTest.java [hunk 2]
```diff
@@ -158,8 +158,65 @@ public void testInsertAfter() {
...
+        testList.insertAtHead(40);
+
+        assertEquals(
+                "should expect 40",
+                40,
+                testList.runningBackwards(5));
+    }
+
+    // test where k is not positive - test should fail and show exception
+    @Test
```
Comment (jeffborda) [inline code401Challenges/src/test/java/code401Challenges/linkedlist/LinkedListTest.java]: This test fails because you are throwing an exception... But you can test that by adding:
@Test(expected=IllegalArgumentException.class)

And then you can remove the assertNull(), just run: testList(runningBackwards(-1)

File: code401Challenges/src/test/java/code401Challenges/linkedlist/LinkedListTest.java [hunk 3]
```diff
@@ -158,8 +158,65 @@ public void testInsertAfter() {
...
+        testList.insertAtHead(50);
+        testList.insertAtHead(10);
+        testList.insertAtHead(30);
+        testList.insertAtHead(60);
+
+        assertNull(testList.runningBackwards(-1));
+    }
+
+    // test where k is greater than the list length - test should fail and show exception
+    @Test
```
Comment (jeffborda) [inline code401Challenges/src/test/java/code401Challenges/linkedlist/LinkedListTest.java]: This test is failing too, so do the same as above.
